packages/amplify-provider-awscloudformation/src/push-resources.ts

```diff
@@ -47,18 +47,24 @@ import { preProcessCFNTemplate } from './pre-push-cfn-processor/cfn-pre-processo
 import { AUTH_TRIGGER_STACK, AUTH_TRIGGER_TEMPLATE } from './utils/upload-auth-trigger-template';
 import { ensureValidFunctionModelDependencies } from './utils/remove-dependent-function';
 import { legacyLayerMigration, postPushLambdaLayerCleanup, prePushLambdaLayerPrompt } from './lambdaLayerInvocations';
+import { storeRootStackTemplate } from './initializer';
+import { transformRootStack } from './override-manager';
+import { Template } from './root-stack-builder/types';
 
 const logger = fileLogger('push-resources');
 
 // keep in sync with ServiceName in amplify-category-api, but probably it will not change
 const ApiServiceNameElasticContainer = 'ElasticContainer';
 
 const spinner = ora('Updating resources in the cloud. This may take a few minutes...');
-const rootStackFileName = 'rootStackTemplate.json';
-const nestedStackFileName = 'nested-cloudformation-stack.yml';
+
 const optionalBuildDirectoryName = 'build';
 const cfnTemplateGlobPattern = '*template*.+(yaml|yml|json)';
 const parametersJson = 'parameters.json';
+export const rootStackFileName = 'rootStackTemplate.json';
+export const nestedStackFileName = FeatureFlags.getBoolean('overrides.project')
```

Comment:
even without feature flag - can we change this file to 'root-cloudformation-stack.json' please? It won't be a breaking. change as this is a runtime file and not checked in.